basic_system/src/system_implementation/flat_storage_model/storage_cache.rs

```diff
@@ -165,7 +167,7 @@ where
                 // correctly.
                 let data_from_oracle = unsafe { dst.assume_init() } ;
 
-                resources_policy.charge_cold_storage_read_extra(ee_type, resources, data_from_oracle.is_new_storage_slot)?;
+                resources_policy.charge_cold_storage_read_extra(ee_type, resources, data_from_oracle.is_new_storage_slot,is_access_list)?;
```

Comment:
Adding `is_access_list` doesn't look beautiful imo (maybe some proper new method would be better). But acceptable if changes aren't worth it